# Exercise: Decision Trees & Combining Methods

In this exercise we will use decision trees and neural networks for classification of Iris species (Setosa, Versicolor, Virginica). For use random_state=0 for splitting and building all models.

1) Fit decision tree with maximum depth (max_depth) of 2 and the default gini index for building the tree. Find the classification accuracy. 
You can visualize the tree using:

       from sklearn.tree import export_graphviz

       import graphviz 

       dot_data=export_graphviz(FittedTreeModel,class_names=iris_dataset.target_names,    feature_names=iris_dataset.feature_names,out_file=None)

       graph = graphviz.Source(dot_data)  

       graph 

2) Use random forests that combines 4 trees, each of maximum depth 2 and maximum number of features considered at each split is 2. What is the classification accuracy?

3) Use AdaBoost with 4 trees to perform the classification. What is the accuracy? Comment on results




In [1]:
%matplotlib inline
from sklearn.datasets import load_iris
import numpy as np
from sklearn.preprocessing import MinMaxScaler 
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import AdaBoostClassifier
from sklearn.neural_network import MLPClassifier


In [5]:
iris = load_iris()
X_train, X_test, Y_train, Y_test = train_test_split(iris['data'],iris['target'],random_state=0)
DecisionTree=DecisionTreeClassifier(random_state=0,max_depth=2)
model1=DecisionTree.fit(X_train,Y_train)
score1=model1.score(X_test,Y_test)
print("Test Score, Decision Tree : ",score1)

Test Score, Decision Tree :  0.894736842105


In [6]:
RandomForest= RandomForestClassifier (n_estimators=4, max_features=2, max_depth=2, random_state=0)
model2=R_Forest.fit(X_train,Y_train)
score2=model2.score(X_test,Y_test)
print("Test Score, Random Forest : ",score2)

Test Score, Random Forest :  0.947368421053


In [8]:
Ada_Boost= AdaBoostClassifier(n_estimators=4, random_state=0)
model3=Ada_Boost.fit(X_train,Y_train)
score3=model3.score(X_test,Y_test)
print("Test Score, ADA Boost : ",score3)

Test Score, ADA Boost :  0.973684210526


ADA boost gives accuracy of 0.973684210526 which is the best accuracy when compared to the other models for IRIS dataset for different petal and sepal lengths.